src/Config.php

```diff
@@ -3750,4 +3750,111 @@ final public static function generateUuid($type)
         self::setConfigurationValues('core', [$type . '_uuid' => $uuid]);
         return $uuid;
     }
+
+    /**
+     * Get email and name to be used as "From" in our email notifications
+     *
+     * @param int  $entities_id
+     * @param bool $allow_response
+     *
+     * @return array [email => sender address, name => sender name]
+     */
+    public static function getAdminEmailSender(
+        int $entities_id,
+        bool $allow_response
+    ): array {
+        if ($allow_response) {
+            return self::getAdminEmail($entities_id);
+        } else {
+            return self::getNoReplyEmail();
+        }
+    }
+
+    /**
+     * Try to find a valid email to use as a sender for notifications
+     * Form first to last, it will look for :
+     * - A valid "from_email" address defined in the global configuration
+     * - A valid "admin_email" address defined in the current entity configuration
+     * - A valid "admin_email" address defined in the global configuration
+     *
+     * @param int $entities_id Current entity
+     *
+     * @return array [email => sender address, name => sender name]
+     */
+    public static function getAdminEmail(int $entities_id): array
+    {
+        global $CFG_GLPI;
+
+        // Read config data for generic from email
+        $from_email      = $CFG_GLPI['from_email']      ?? "";
+        $from_email_name = $CFG_GLPI['from_email_name'] ?? "";
+
+        // Check if generic from email is valid
+        if (NotificationMailing::isUserAddressValid($from_email)) {
+            return [
+                'email' => $from_email,
+                'name'  => $from_email_name,
+            ];
+        }
+
+        // First fallback: check admin email in entity
+        $entity_admin_email = trim(
+            Entity::getUsedConfig('admin_email', $entities_id, '', '')
+        );
+        $entity_admin_email_name = trim(
+            Entity::getUsedConfig('admin_email_name', $entities_id, '', '')
+        );
+
+        // Check if admin email for the current entity is valid
+        if (NotificationMailing::isUserAddressValid($entity_admin_email)) {
+            return [
+                'email' => $entity_admin_email,
+                'name'  => $entity_admin_email_name,
+            ];
+        }
+
+        // Second fallback: read config for the global admin email
+        $admin_email      = $CFG_GLPI['admin_email']      ?? "";
+        $admin_email_name = $CFG_GLPI['admin_email_name'] ?? "";
+
+        // Check if admin email is valid
+        if (NotificationMailing::isUserAddressValid($admin_email)) {
+            return [
+                'email' => $admin_email,
+                'name'  => $admin_email_name,
+            ];
+        }
+
+        // Final fallback, no valid values found
+        return [
+            'email' => '',
+            'name'  => '',
```

Comment:
Previous version was sending null, which seems more appropriate to me.
```suggestion
            'email' => null,
            'name'  => null,
```